packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +586,60 @@ export namespace meeting {
       sendMessageToParent('meeting.appShareButton.setOptions', [shareInformation]);
     }
   }
+
+  /**
+   * Ext App will handle the Audio(Mic & Speaker). Teams to go Audio less (No Audio)
+   *
+   * @throws standard parameter validation error
+   * @param isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *    false - Audio transferred to Teams and Teams to restore the audio.
+   * @param callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains IRequestAppAudioHandlingSdkResponse param, error or operation result.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    callback: (
+      error: SdkError | null,
+      requestAppAudioHandlingSdkResponse: IRequestAppAudioHandlingSdkResponse | null,
+    ) => void,
+    isAppHandlingAudio: boolean,
+    callbackMicMuteStateChangedHandler: (micState: IMicState) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[requestAppAudioHandling] Callback response cannot be null');
+    }
+    if (!callbackMicMuteStateChangedHandler) {
+      throw new Error('[requestAppAudioHandling] Callback Mic mute state handler cannot be null');
+    }
+    ensureInitialized(runtime, FrameContexts.sidePanel, FrameContexts.meetingStage);
+
+    const callbackInternal = (
+      error: SdkError | null,
+      requestAppAudioHandlingSdkResponse: IRequestAppAudioHandlingSdkResponse | null,
+    ): void => {
+      callback(error, requestAppAudioHandlingSdkResponse);
+      if (!error) {
+        registerHandler('meeting.micStateChanged', callbackMicMuteStateChangedHandler);
```

Comment:
I think it would be better to do this before invoking the callback. Also, you need to clear the handler when a app stops handling the audio.